Add blog article “Securing Admission Controllers” (#31221)
kubernetes/website: content/en/blog/_posts/Securing-Admission-Controllers.md

---
layout: blog
title: "Securing Admission Controllers"
date: YYYY-MM-DD

> slug: text-for-URL-link-here-no-spaces

Reviewer: How about:
```suggestion
slug: secure-your-admission-controllers-and-webhooks
```
?

---

> # Securing Admission Controllers

Reviewer: This is the title, you should omit it from the Markdown (it will be rendered in from the front matter).

Author: removed.

> Admission control is a key part of Kubernetes security, alongside authentication and authorization. Webhook admission controllers are extensively used to help improve the security of Kubernetes clusters in a variety of ways including restricting the privileges of workloads and ensuring that images deployed to the cluster meet organization’s security requirements.

Reviewer: ```suggestion
[Admission control](/docs/reference/access-authn-authz/admission-controllers/) is a key part of Kubernetes security, alongside authentication and authorization. Webhook admission controllers are extensively used to help improve the security of Kubernetes clusters in a variety of ways including restricting the privileges of workloads and ensuring that images deployed to the cluster meet organization’s security requirements.
```

> However, as with any additional component added to a cluster, there are security risks which can present themselves, if the deployment and management of the admission controller isn’t handled correctly.  To help admission controller users and designers manage these risks appropriately, SIG-Security docs have spent some time developing a threat model for admission controllers. This threat model looks at likely risks which may arise from the incorrect use of admission controllers, which could allow security policies to be bypassed, or even allow an attacker to get unauthorised access to the cluster.

Reviewer: ```suggestion
However, as with any additional component added to a cluster, there are security risks which can present themselves, if the deployment and management of the admission controller isn’t handled correctly.  To help admission controller users and designers manage these risks appropriate, the [security documentation](https://github.com/kubernetes/community/tree/master/sig-security#security-docs) subgroup of SIG Security have spent some time developing a threat model for admission controllers. This threat model looks at likely risks which may arise from the incorrect use of admission controllers, which could allow security policies to be bypassed, or even allow an attacker to get unauthorised access to the cluster.
```

From the threat model, we developed a set of security best practices which should be adopted to ensure that cluster operators can get the security benefits of admission controllers whilst avoiding any risks from using them.

> ## Admission Controller Best Practices

Reviewer: I prefer “good practice” instead of the term “best practice”, so:
```suggestion
## Admission controllers and good practices for security
```

* **Admission controller fails closed**. This is a security practice that has a tradeoff, so whether a cluster operator wants to configure it will depend on the cluster’s threat model. If an admission controller fails closed, when the API server can’t get a response from it, all deployments will fail. This stops attackers bypassing the admission controller by disabling it, but obviously can disrupt the cluster’s operation. As clusters can have multiple webhooks, one approach to hit a middle ground might be to have critical controls on a fail closed setups and less critical controls allowed to fail open.
* **Regular reviews of webhook configuration**. Configuration mistakes can lead to security issues, so it’s important that the admission controller webhook configuration is checked to make sure the settings are correct. This kind of review could be done automatically by an Infrastructure As Code scanner or manually by an administrator.

> ### Secure Cluster Configuration

Reviewer: ```suggestion
### Secure cluster configuration for admission control
```

* **Restrict RBAC rights**. Any user who has rights which would allow them to modify the configuration of the webhook objects or the workload that the admission controller uses, could disrupt its operation. So it’s important to make sure that only cluster administrators have those rights.
* **Prevent Privileged Workloads**. One of the realities of container systems is that if a workload is given certain privileges, it will be possible to break out to the underlying cluster node and impact other containers on that node. Where admission controller services run in the cluster they’re protecting, it’s important to ensure that any requirement for privileged workloads is carefully reviewed and restricted as much as possible.
* **Strictly Control External System Access**. As a security service in a cluster admission controller systems will have access to sensitive information like credentials. To reduce the risk of this information being sent outside the cluster, network policies should be used to restrict the admission controller services access to external networks.

> * **Each Cluster has a dedicated webhook**. Whilst it may be possible to have admission controller webhooks which serve multiple clusters, there is a risk when using that model that an attack on the webhook service would have a larger impact where it’s shared. Also where multiple clusters use an admission controller there will be increased complexity and access requirements, making it harder to secure.

Reviewer: nit: make all the bold text sentence case. Title case could work too; if so, make sure it's consistent.

Author: changed to sentence case, think I got them all :)

> ### Admission Controller Rules

Reviewer: (nit)
```suggestion
### Admission controller rules
```
